### PR TITLE
Use correct tomcrypt headers

### LINKED ISSUE #1885: Support recent system library tomcrypt
In a similar vein as #1881, there are problems when the tomcrypt headers and libraries are not matching.

So with `WITH_SYSTEM_TOMCRYPT=ON`, the bundled headers are being compiled against but the system libraries of a different version are being linked against leading to `SIGABRT`:
```gdb
#0  0x00007ffff4c4d755 in raise () from /usr/lib/libc.so.6
#1  0x00007ffff4c38851 in abort () from /usr/lib/libc.so.6
#2  0x00007ffff4c8fa38 in __libc_message () from /usr/lib/libc.so.6
#3  0x00007ffff4c9625a in malloc_printerr () from /usr/lib/libc.so.6
#4  0x00007ffff4c98f2c in _int_malloc () from /usr/lib/libc.so.6
#5  0x00007ffff4c9a7ff in malloc () from /usr/lib/libc.so.6
#6  0x00007ffff4fdaaca in operator new (sz=2376)
    at /build/gcc/src/gcc/libstdc++-v3/libsupc++/new_op.cc:50
#7  0x0000555555ca0e9b in sm_main (argc=1, argv=0x7fffffffe008)
    at ../src/StepMania.cpp:1170
#8  0x0000555556422073 in main (argc=1, argv=0x7fffffffe008)
    at ../src/Main.cpp:5
```

### PR DESCRIPTION
**Pull request by mkroening**
Fixes #1885.

### CODE REVIEW

File: src/CryptManager.cpp [hunk 1]
```diff
@@ -80,7 +80,6 @@ static const int KEY_LENGTH = 1024;
  */
 
 static PRNGWrapper *g_pPRNG = nullptr;
-ltc_math_descriptor ltc_mp = ltm_desc;
```
Comment (mkroening) [inline src/CryptManager.cpp]: I'll need to have another look at this.

Comment (mkroening) [reply src/CryptManager.cpp]: This statically sets the used math descriptor. See 17.7.19 in [crypt-1.18.2.pdf](https://github.com/libtom/libtomcrypt/releases/download/v1.18.2/crypt-1.18.2.pdf)

**Comment by mkroening** [comment]
These changes are quite straight forward. Similar to https://github.com/stepmania/stepmania/pull/1854#discussion_r443122332 this makes sure, we compile against system headers if we link against system libraries.

**Comment by quietly-turning** [comment]
This seems fine. 👍 

There's currently a simple conflict in ci.yml because the libpng upgrade in #1854 got merged first.  Sorry to have you rebase and force-push one more time.

---

Does anyone have experience with GitHub's web editor for resolving conflicts in pull requests?  Is it okay to use here?

I tried [fetching this pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally) to a local branch, rebasing, resolving, and pushing to upstream, but I ended up creating a new branch (as GitHub's guide states will happen) rather than adding to this.  

On the plus side, the relevant CI has already passed: https://github.com/stepmania/stepmania/actions/runs/148110342

**Comment by geefr** [comment]
Haven't encountered github's one but in my experience using the web editors is a good way to break things, or have messy commit logs as it probably has a weird workflow compared to editing on disk (If you're using bitbucket the web editor can even bypass pull requests/commit hooks entirely, force update master, and permanently brick your history. Expect github's is a bit better than that at least..).

Probably fine in this case though so could try it, worst case you've got to roll back history or force push the changes from a local repo again to recover.

(btw it's great to see progress on sm recently, especially on maintenance/documentat. Keep up the great work, watch out for burnout ;D)

**Comment by quietly-turning** [comment]
> Haven't encountered github's one but in my experience using the web editors is a good way to break things

Yeah, that's what I'd heard as well, but it'd been a few years since I checked.  

---

> Probably fine in this case though so could try it, worst case you've got to roll back history or force push the changes from a local repo again to recover.

It seems to have worked all right here.  Thanks for the suggestion.  And thanks for contributing this, @mwkroening!

---

> (btw it's great to see progress on sm recently, especially on maintenance/documentat. Keep up the great work, watch out for burnout ;D)

Thanks go to all the contributors putting in time making and testing changes, submitting pull requests, and engaging in dialogue here recently. 🙂

(Your still-open #1567 is on the to-review-soon list.)

**Pull request merged**